x/stakeibc/keeper/hooks.go

```diff
@@ -92,9 +92,7 @@ func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochInfo epochstypes.EpochInf
 	// At the hour epoch, query the swap price on each trade route and initiate the token swap
 	if epochInfo.Identifier == epochstypes.HOUR_EPOCH {
 		k.UpdateAllSwapPrices(ctx)
```

Comment:
should we be removing this too? the authz grantee could just query the price offchain right before swapping rather than using on-chain metric that is more stale

Reply:
I don't feel strongly either way.

My thinking was I wanted to make the changes as minimal as possible, including not modifying the trade route structs (so we don't need to do a migration). And if we're going to leave the price on the struct, it felt like we may as well keep it updated. 

That said, I also don't love leaving in dead code so can see an argument for removing it

Reply:
You know what - maybe we should just leave it. It's working now and will act as a backup price - you never know why we may need it. Given the current upgrade timeline I agree with you philosophy of `make the changes as minimal as possible`

Reply:
To sum up: don't feel that strongly either way, slightly leaning towards keeping this as is. 

Leaving this open for comments from @shellvish @asalzmann @ethan-stride.

Reply:
Ethan's comment from Slack: _We don’t need it any longer and it does fire once an hour which is a lot. I would only keep it in if we thought that after the community pool comes through we might go back to the auto trade on chain approach._

Reply:
yeah the more I think about it, I think I actually lean towards pulling it out. I think a lot of the trade route struct will no longer be relevant (and we can migrate it later to reflect that)

It's pretty simple to remove and leaves one less place for something to break

Reply:
If it's easy to pull out with no migrations then let's do it! I got the impression from above that it might require a migration?

Reply:
oh sorry to clarify the options are:
* (1) Leave query and struct as is
* (2) Turn off the query but _keep_ the price and last updated time on the struct
* (3) Turn off the query and _remove_ the price and last updated time from the struct (requires migration)

My argument originally was that #2 felt a bit odd to turn off the updates but leave a stale value on the struct. But on second thought, I think I'd prefer #2 and then we can do the migration in a later upgrade to clean it up

Reply:
Thanks for writing these out - number 2 feels right!

Reply:
#2 makes sense - why does #3 require a migration? I'd have assumed we'd just mark the proto field as deprecated, which doesn't require a migration afaik.

Reply:
that's true, probably wouldn't require a migration

Reply:
If no migration is required, 3 seems preferable to 2, although I realize we're getting away from the principle of making minimal changes

Reply:
I think 3 would still require us to update all references to those fields (otherwise we'll get link errors)

I agree that's the best long term solution but I think I'd prefer to push the cleanup to a separate upgrade to make sure we can meet our timeline

Reply:
@riley-stride just turned the price query in the latest commit

Reply:
Do we make any assumptions about the price being non-nil? If you're confident we don't, this lgtm and I'm signed off. Approving now.

Reply:
It wont be nil - it'll just be stale.

Also, a nil price would just lead to us disabling the on-chain swap (which we already are doing!)